### ISSUE TITLE
Extremely slow response time from Prompts

### ISSUE THREAD

**Issue by Bigcow11** [antimatter15/alpaca.cpp#118]
Has anyone else had issues with the response taking forever on alpaca? Have the 13B version installed and operational; however, when prompted for an output the response is extremely slow. 

For example: 5+ minutes to output text responding to the sample question of: "What are lists in python?". This is also apparent through terminal so not isolated to the web GUI.

Hardware is more than sufficient as well to run it
![alpacaspeed](https://user-images.githubusercontent.com/38636346/226862709-b8129b21-e5ab-4505-8f20-5d94c9a4c53c.png)

**Comment by DartPower** [comment]
Also have this issue, but IDK this is normal or not.

**Comment by freezah** [comment]
Same here. Running on 16 Core AMD CPU and is extremely slow.

**Comment by trevtravtrev** [comment]
What hardware? Likely the only issue I can think of. I'm running 7B, 13B, and 30B on 32GB RAM and a beefy cpu. 7B is snappy, 13B is still fast, 30B takes a couple minutes to output the full answers.

**Comment by freezah** [comment]
@trevtravtrev 
I’m using AMD 5950X (32 threads) with 128GB of RAM. Tried with 7B as well and also super slow. Even I run the “chat” with -t 32 parameter, I see mostly one thread being used at 100% and rest is kind of idling.

**Comment by pnadj** [comment]
Same with the basic 7B model

**Comment by ninp0** [comment]
+1 debian VM, 16 core & 64 gb memory

**Comment by KaruroChori** [comment]
> 

I would suggest to use 30 threads, not 32. It is better to keep a few available for other processes.

**Comment by KaruroChori** [comment]
@freezah
I would suggest you using 30 threads, not 32. It is better to keep a few available for other processes, based on your system load.

**Comment by freezah** [comment]
@KaruroChori 
I totally agree. I used 32 just to verify if the application will work better than with default 4 threads.

I just run it with 30 threads and observing same issue. Even though I see 30 forked processes in 'htop', only one of them consumes 100% of single thread, rest is consuming 3.3-4.0% on average.

**Comment by KaruroChori** [comment]
I tested both the 7b and the 30b on a xeon 2650Lv4  128GB of ram and a ryzen 5950x with 64GB of ram. Performance are comparable between the two platform (between 2-3 times faster for the ryzen)
I am not sure why some people are getting an extremely slow generation.
When the process is running, all cores should be around 100%, not just one.

**Comment by freezah** [comment]
@KaruroChori 

Which host system distro, version, kernel you are using ?

**Comment by KaruroChori** [comment]
My uname -a:
Linux [redacted] 5.15.0-40-generic #43-Ubuntu SMP Wed Jun 15 12:54:21 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
Linux [redacted] 5.15.0-67-generic #74-Ubuntu SMP Wed Feb 22 14:14:39 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

**Comment by kaiserfr** [comment]
Same here. 
Have an AMD A10-6700 CPU and 32GB RAM. I try to run it on Ubuntu Ubuntu 20.04.6 and WSL2.
My guess is that some speed optimizations like AVX or NEON do not work with AMD processors.

Do AMD processers have other, similar speedup flags, which must be used instead?

**Comment by Santiagoc57** [comment]
Same problem here

**Comment by hunterdrayman** [comment]
I am getting this too with 20 cores on twin Xeon x5675s and 12Gb of memory. This btop screen dump suggests it's CPU bound and that disk & memory aren't the cause of the bottleneck. So, given the claims that it runs happily on a laptop with no GPU I'm assuming something is amiss. 

![image](https://user-images.githubusercontent.com/72385639/227924979-9de03d2e-c405-472a-b5df-a117653b1da7.png)

**Comment by hunterdrayman** [comment]
Just for a laugh I asked Google Bard how many cores are needed to run a 7B model:

> A 7B model like Alpaca would need a cluster of multiple cores to run at a good conversational speed. The exact number of cores would depend on the specific hardware and software configuration, but it would likely be in the range of hundreds or even thousands of cores.

>This is because 7B models are very large and complex, and they require a lot of processing power to run. Even with a powerful computer, it can take several minutes for a 7B model to generate a single response.

>However, the speed of a 7B model is not the only important factor. The quality of the responses is also important, and 7B models are known to generate very high-quality responses. This makes them a good choice for applications where accuracy and fluency are important, such as customer service or education.
